serverless.rb

```diff
@@ -57,13 +53,14 @@
 
 # puts "Trimmed bucket name to not include stage suffix: #{bucket_name}"
 
-s3_name = 'S3Bucket' + bucket_name.gsub('-', '').capitalize
+s3_name = 'S3Bucket' + bucket_name.gsub('${self:provider.stage}', stage).gsub(/[-.]/, '').capitalize
+s3_name_no_env = 'S3Bucket' + bucket_name.gsub('${self:provider.stage}', '').gsub(/[-.]/, '').capitalize
```

Comment:
@stretch1414 So my Ruby bullshit is still necessary because of the way serverless generates s3 buckets when it builds cloudformation. The bucket create is actually caused by: 
```
  startExecution:
    handler: lambda/lambda.startExecution
    events:
      - s3:
          bucket: ${self:custom.imageUploadBucket}
          event: s3:ObjectCreated:*
          rules:
            - prefix: raw/
    memorySize: 128
    environment:
      STATEMACHINE_ARN: "arn:aws:states:#{AWS::Region}:#{AWS::AccountId}:stateMachine:${self:custom.thumbnailProcessorName}"

```

which creates the bucket automatically, but we need to edit properties of that bucket and in order to do that we have to use the same cloudformation key. Otherwise there's no way to tell cf that a second part of the bucket configuration depends on the first part if we don't know the name of the resource for the first part. Chicken and egg bullshit.

Reply:
Makes sense.  Fucky, but makes sense.